src/controllers/auth.js

```diff
@@ -1,2 +1,62 @@
 const bcrypt = require('bcrypt');
 const UserModel = require('../models/user');
+const { generateUniqeUsername } = require('../services/utils');
+
+// eslint-disable-next-line consistent-return
+const register = async (req, res) => {
+  const errorsArray = [];
+  const {
+    firstName,
+    lastName,
+    email,
+    phoneNumber,
+    age,
+    gender,
+    nationality,
+    refugee,
+    password0,
+  } = req.body;
+
+  try {
+    const usedEmail = await UserModel.findOne({ email });
+    if (usedEmail) {
+      errorsArray.push('Email is already taken');
+    }
+
+    if (errorsArray.length > 0) {
+      return res.status(400).json({ error: errorsArray });
+    }
+
+    const passwordHash = await bcrypt.hash(password0, 10);
+    const newUser = await UserModel.create({
+      firstName,
+      lastName,
+      email,
+      username: generateUniqeUsername(email),
+      phoneNumber,
+      age,
+      gender,
+      nationality,
+      refugee,
+      provider: 'Local',
+      providerId: 'Local',
+      passwordHash,
+    });
+
+    return res.status(201).json({ created: newUser });
```

Comment:
here lets return the newUser right away, no need to nest it insside created key

Reply:
plus I dont think its safe to return the newUser with all the data that it has with it